oap-server/server-configuration/configuration-configmap/src/main/java/org/apache/skywalking/oap/server/configuration/configmap/ConfigmapConfigurationWatcherRegister.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.configuration.configmap;
+
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import java.util.Optional;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.oap.server.configuration.api.ConfigTable;
+import org.apache.skywalking.oap.server.configuration.api.ConfigWatcherRegister;
+
+@Slf4j
+public class ConfigmapConfigurationWatcherRegister extends ConfigWatcherRegister {
+
+    private final ConfigurationConfigmapInformer informer;
+
+    public ConfigmapConfigurationWatcherRegister(ConfigmapConfigurationSettings settings,
+                                                 ConfigurationConfigmapInformer informer) {
+        super(settings.getPeriod());
+        this.informer = informer;
+    }
+
+    @Override
+    public Optional<ConfigTable> readConfig(Set<String> keys) {
+        final ConfigTable configTable = new ConfigTable();
+
+        Optional<V1ConfigMap> v1ConfigMap = informer.configMap();
+
+        for (final String name : keys) {
+
+            final String value = v1ConfigMap.map(configMap -> configMap.getData().get(name)).orElse(null);
```

Comment:
If you can't read the key from configmap, please don't include value in the ConfigTable. I received a report today, someone accidentally reset the default value, even they didn't config the new value in the dynamic configuration center.

Reply:
ok,thanks for your notice